src/devices/cpu/tlcs900/tmp94c241.cpp

```diff
@@ -0,0 +1,1520 @@
+	{
+		/*** NOTE: TIMER_CHANGE_8 is stored at m_timer_change[6] ***/
+
+		switch( m_t8mod & 3 ) /* T8_INPUT_CLOCK */
+		{
+		case 0:  /* TIA */
+			// TODO: implement-me!
+			break;
+		case 1:  /* T1 */
+			m_timer_change[6] += ( m_timer_pre >> 3 ) - ( old_pre >> 3 );
+			break;
+		case 2:  /* T4 */
+			m_timer_change[6] += ( m_timer_pre >> 5 ) - ( old_pre >> 5 );
+			break;
+		case 3:  /* T16 */
+			m_timer_change[6] += ( m_timer_pre >> 7 ) - ( old_pre >> 7 );
+			break;
+		}
+
+		for( ; m_timer_change[6] > 0; m_timer_change[6]-- )
+		{
+			m_timer_16[2]++; /* UPCOUNTER_8 */
+			if ( ((m_timer_16[2] == m_treg_16[TREG9]) && BIT(m_t8ffcr, 3)) ||
+				((m_timer_16[2] == m_treg_16[TREG8]) && BIT(m_t8ffcr, 2)) )
+			{
+				change_timer_flipflop( 8, FF_INVERT );
+				m_timer_16[2] = 0;
+				m_int_reg[INTET89] |= 0x08;
+				m_check_irqs = 1;
+			}
+		}
+	}
+
+	if ( BIT(m_t16run, 3) ) /* Timer A is running */
+	{
+		/*** NOTE: TIMER_CHANGE_A is stored at m_timer_change[7] ***/
+
+		switch( m_tamod & 3 ) /* TA_INPUT_CLOCK */
+		{
+		case 0: /* TIA */
+			// TODO: implement-me!
+			break;
+		case 1: /* T1 */
+			m_timer_change[7] += ( m_timer_pre >> 3 ) - ( old_pre >> 3 );
+			break;
+		case 2: /* T4 */
+			m_timer_change[7] += ( m_timer_pre >> 5 ) - ( old_pre >> 5 );
+			break;
+		case 3: /* T16 */
+			m_timer_change[7] += ( m_timer_pre >> 7 ) - ( old_pre >> 7 );
+			break;
+		}
+
+		for( ; m_timer_change[7] > 0; m_timer_change[7]-- )
+		{
+			m_timer_16[3]++; /* UPCOUNTER_A */
+			if ( ((m_timer_16[3] == m_treg_16[TREGA]) && BIT(m_taffcr, 2)) ||
+				((m_timer_16[3] == m_treg_16[TREGB]) && BIT(m_taffcr, 3)) )
+			{
+				change_timer_flipflop( 0xa, FF_INVERT );
+				m_timer_16[3] = 0;
+				m_int_reg[INTETAB] |= 0x08;
+				m_check_irqs = 1;
+			}
+		}
+	}
+
+	m_timer_pre &= 0xffffff;
+}
+
+
+//-------------------------------------------------
+//  execute_set_input - called when a synchronized
+//  input is changed
+//-------------------------------------------------
+
+void tmp94c241_device::execute_set_input(int input, int level)
+{
+	switch( input )
+	{
+	case INPUT_LINE_NMI:
+	case TLCS900_NMI:
+		if ( level != m_level[TLCS900_NMI])
+		{
+			m_level[TLCS900_NMI] = level;
+			if (level == ASSERT_LINE)
+				m_nmi_state = ASSERT_LINE;
+		}
+		break;
+
+	case TLCS900_INTWD:
+		break;
+
+	case TLCS900_INT0:
+		if (m_iimc & 0x02)
+		{
+			/* Rising edge detect */
+			if (level != m_level[TLCS900_INT0] && level == ASSERT_LINE)
+			{
+				/* Leave HALT state */
+				m_halted = 0;
+				m_int_reg[INTE0AD] |= 0x08;
+			}
+		}
+		else
+		{
+			/* Level detect */
+			if (level == ASSERT_LINE)
+				m_int_reg[INTE0AD] |= 0x08;
+			else
+				m_int_reg[INTE0AD] &= ~0x08;
+		}
+		m_level[TLCS900_INT0] = level;
+		break;
+
+	case TLCS900_INT4:
+		if (level != m_level[TLCS900_INT4]) {
+			m_level[TLCS900_INT4] = level;
+			if (level == ASSERT_LINE)
+				m_int_reg[INTE45] |= 0x08;
+			else
+				m_int_reg[INTE45] &= ~0x08;
+		}
+		break;
```

Comment:
Similarly, there are eight almost identical copies of this.  Can you make a local lamba that takes the index to `m_level`, index to `m_int_reg` and bit mask for `m_int_reg` as parameters?

Reply:
Sure! Do you like it this way? https://github.com/mamedev/mame/pull/13220/commits/38610081dc0d230795c465194f4fda069ef0a7fb